platform/android/src/jni.cpp

```diff
@@ -757,7 +758,26 @@ jni::jarray<jlong>* nativeAddPolygons(JNIEnv *env, jni::jobject* obj, jlong nati
         jni::jobject* polygon = jni::GetObjectArrayElement(*env, *jarray, i);
         jni::jobject* points = jni::GetField<jni::jobject*>(*env, polygon, *polygonPointsId);
 
-        mbgl::FillAnnotation annotation { mbgl::Polygon<double> { toGeometry<mbgl::LinearRing<double>>(env, points) } };
+        mbgl::Polygon<double> geometry { toGeometry<mbgl::LinearRing<double>>(env, points) };
+
+        jni::jobject* holes = jni::GetField<jni::jobject*>(*env, polygon, *polygonHolesId);
+        NullCheck(*env, holes);
+        jni::jarray<jni::jobject>* jarrayHoles =
+            reinterpret_cast<jni::jarray<jni::jobject>*>(jni::CallMethod<jni::jobject*>(*env, holes, *listToArrayId));
+        NullCheck(*env, jarrayHoles);
+
+        std::size_t size = jni::GetArrayLength(*env, *jarrayHoles);
+        for (std::size_t j = 0; j < size; j++) {
+            jni::jobject* hole = reinterpret_cast<jni::jobject*>(jni::GetObjectArrayElement(*env, *jarrayHoles, j));
```

Comment:
This cast is redundant, `jni::GetObjectArrayElement` returns `jni::jobject*` already